### ISSUE TITLE
Spurious warning when using non-ASCII characters in function names 

### ISSUE THREAD

**Issue by cpitclaudel** [PyCQA/pylint#1176]
With the following `test.py`:

```
def パイソン():
    print("パイソン!")

パイソン()

def différence(x, y):
    pass
```

I get this:

```
$ python3 -m pylint "-r" "n" test.py 
No config file found, using default configuration
************* Module test
C:  1, 0: Invalid function name "パイソン" (invalid-name)
C:  6, 0: Invalid function name "différence" (invalid-name)
```

I can see why it may not be ideal to use non-ascii characters in function names, but maybe "invalid" is a bit of a strong word?

**Comment by PCManticore** [comment]
pylint is expecting the names to functions and similar objects to be in ASCII. You can control these checks with the corresponding regular expressions from the config file. Not sure what is wrong with the *invalid* word.

**Comment by cpitclaudel** [comment]
> pylint is expecting the names to functions and similar objects to be in ASCII. You can control these checks with the corresponding regular expressions from the config file.

I'm aware of this (but thanks!)

> Not sure what is wrong with the **invalid** word.

It just doesn't sound right.  The function name is valid as far as python is concerned.  Contrast with other uses of invalid in pylint:

> Using invalid lookup key 1 in format specifier

> Invalid format string

> Invalid object in \_\_all\_\_

> Invalid metaclass %r used

> Invalid characters %r in a docstring

> Invalid \_\_slots\_\_ object

> Invalid object %r in \_\_slots\_\_, must contain…

Do any of these describe style warnings?  "Invalid" just evokes an error to me.

**Comment by PCManticore** [comment]
That makes sense. What synonym would you use, though?

**Comment by cpitclaudel** [comment]
I can't think of a good synonym but maybe the following rephrasing could help?

    C:  1, 0: Function name "パイソン" doesn't conform to 'function-rgx' template (invalid-name)

Or this?

    C:  6, 0: Function name "パイソン" doesn't match 'function-rgx' template (invalid-name)

Or, if the regexp hasn't been customized:

    C:  6, 0: Function name "パイソン" isn't PEP8-conformant (invalid-name)

**Comment by The-Compiler** [comment]
I like something like "doesn't match 'function-rgx' setting" or so (though "template" is fine as well) - that also makes it clear that it can be customized.

**Comment by PCManticore** [comment]
Sounds better indeed, thanks! If you want, feel free to submit a patch.

**Comment by rogalski** [comment]
I'm not sure what are the plans for #1046, but it may affect a decision on what message should be picked.
